### PR TITLE
expose parcoords context line color

### LINKED ISSUE #3939: Add option to set parcoords context line color and opacity
The deselected lines in all `parcoords` plots now use a constant color (i.e. `#777` and auto opacity logic (i.e. depending on the number of overlaid lines).
In some scenarios it may be useful to let the user override the default.

### PR DESCRIPTION
**Pull request by archmoj**
resolves #3939 | [Demo](https://codepen.io/MojtabaSamimi/pen/eYNMNvR?editors=0010)
by adding new attribute `unselected.line.color`.

TODO: 
- [x] finalize attribute name
- [x] add test

@plotly/plotly_js 
cc: @chriddyp

### CODE REVIEW

File: src/traces/parcoords/attributes.js [hunk 1]
```diff
@@ -141,6 +141,21 @@ module.exports = {
         description: 'The dimensions (variables) of the parallel coordinates chart. 2..60 dimensions are supported.'
     }),
 
+    bgline: {
+        color: {
```
Comment (alexcjohnson) [inline src/traces/parcoords/attributes.js]: If we follow the example of `scatter` we get `unselected.line.color`. It seems a bit overly-nested, though `splom` does this too and all it has in `unselected` is `marker`. I think we should do it, gives us maximum extensibility as well as matching other trace types.

Comment (archmoj) [reply src/traces/parcoords/attributes.js]: Good call. Done in https://github.com/plotly/plotly.js/pull/4635/commits/9b4711b74cd5c30dfccdce3ef27b875c2856f65a.

File: src/traces/parcoords/attributes.js [hunk 2]
```diff
@@ -141,6 +141,23 @@ module.exports = {
...
+                role: 'style',
+                editType: 'plot',
+                description: [
+                    'Sets the color of lines in the background',
+                    'i.e. unselected lines.'
+                ].join(' ')
+            },
+            editType: 'calc'
+        },
+        editType: 'calc'
```
Comment (alexcjohnson) [inline src/traces/parcoords/attributes.js]: if `color` has `editType: 'plot'` and it's the only thing in these containers, they can have `'plot'` as well.

Comment (archmoj) [reply src/traces/parcoords/attributes.js]: I recall, Etienne mentioned that for containers we should use `calc`.

File: test/jasmine/tests/parcoords_test.js [hunk 3]
```diff
@@ -910,6 +922,62 @@ describe('parcoords Lifecycle methods', function() {
...
+
+            return Plotly.restyle(gd, 'unselected.line.color', '#00F');
+        })
+        .then(function() {
+            var rgb = getAvgPixelByChannel(testLayer);
+            expect(rgb[0]).toBe(0, 'no red');
+            expect(rgb[1]).toBe(0, 'no green');
+            expect(rgb[2]).not.toBe(0, 'blue');
+
+            return Plotly.restyle(gd, 'unselected.line.color', 'rgba(0,0,0,0)');
```
Comment (alexcjohnson) [inline test/jasmine/tests/parcoords_test.js]: Let's turn one of these `restyle` calls into a `react` call.

Comment (archmoj) [reply test/jasmine/tests/parcoords_test.js]: If you don't mind, I'd be happy to keep this one, and add a new one to use `react`.

Comment (archmoj) [reply test/jasmine/tests/parcoords_test.js]: Done in https://github.com/plotly/plotly.js/pull/4635/commits/1af1c90bf445b37d2ddab8e1942358fb5ab2d3c2.

File: src/traces/parcoords/attributes.js (hunk 2 above)
Comment (alexcjohnson) [reply src/traces/parcoords/attributes.js]: I believe containers should use the most intense `editType` of anything inside them, and should use `calc` if there's any array or `arrayOk` attribute inside.

File: test/jasmine/tests/parcoords_test.js (hunk 3 above)
Comment (alexcjohnson) [reply test/jasmine/tests/parcoords_test.js]: That works for me!

File: src/traces/parcoords/attributes.js (hunk 2 above)
Comment (archmoj) [reply src/traces/parcoords/attributes.js]: Thanks for the note regarding `arrayOk` items.
Changed all to `plot` in https://github.com/plotly/plotly.js/pull/4635/commits/b4dabb224c37724951f638462dcb1421ab85011e.

**Pull request merged**